File: TUTORIAL_4/WORKSPACE/pyHelloComp/src/acstut/HelloComponentImpl.py
import acstutorial
import acstutorial__POA
import customtypes
import customtypes__POA
  
from Acspy.Servants.ACSComponent import ACSComponent
from Acspy.Servants.ContainerServices import ContainerServices
from Acspy.Servants.ComponentLifecycle import ComponentLifecycle
  
class HelloComponentImpl(acstutorial__POA.HelloComponent, ACSComponent, ContainerServices, ComponentLifecycle):
    
    def __init__(self):
        ACSComponent.__init__(self)
        ContainerServices.__init__(self)
        self._logger = self.getLogger()
    
    def printHi(self):
        print("Just printing 'Hello World!'")
        return

    def getPosition(self):
        p = customtypes.Position(10.0, 5.4)
        return p


    def computeDistance(self, p1, p2):
        return 15.5
    
    def computeCenterOfMass(self):
        return 16.7

File: TUTORIAL_4/WORKSPACE/jHelloComp/src/acsws/acstutorial/HelloComponentImpl/HelloComponentImpl.java
package acsws.acstutorial.HelloComponentImpl;
import alma.acs.component.ComponentImplBase;
import acsws.acstutorial.HelloComponentOperations;
import java.lang.Math;
import acsws.customtypes.Position;


public class HelloComponentImpl extends ComponentImplBase implements HelloComponentOperations {
    public HelloComponentImpl() {
    }
    
    
    public void printHi() 
    {
        System.out.println("Hello World!");
        m_logger.info("hello world!!");
    }

    public Position getPosition()
    {
        Position p = new Position(15.4, 12.4);
        m_logger.info("New position: " + p.az + ", " + p.el);
        return p;
        
    }

    public double computeDistance(Position p1, Position p2)
    {
        double distance = 0;
        distance = Math.sqrt(Math.pow((p2.az - p1.az), 2) + Math.pow((p2.el - p1.el), 2));

        return distance;
    }

    public double computeCenterOfMass()
    {
        return 42;
    }

}